src/structures.jl

```diff
@@ -55,17 +55,77 @@ end
 mutable struct TulipaConstraint
     indices::DataFrame
     table_name::String
+    num_rows::Int
+    constraint_names::Vector{Symbol}
     expressions::Dict{Symbol,Vector{JuMP.AffExpr}}
+    duals::Dict{Symbol,Vector{Float64}}
 
     function TulipaConstraint(connection, table_name::String)
         return new(
             DuckDB.query(connection, "SELECT * FROM $table_name") |> DataFrame,
             table_name,
+            only([ # only makes sure that a single value is returned
+                row.num_rows for
+                row in DuckDB.query(connection, "SELECT COUNT(*) AS num_rows FROM $table_name")
+            ]), # This loop is required to access the query resulted, because it's a lazy struct
+            Symbol[],
+            Dict(),
             Dict(),
         )
     end
 end
 
+"""
+    attach_constraint!(model, cons, name, container)
+
+Attach a constraint named `name` stored in `container`, and set `model[name] = container`.
+This checks that the `container` length matches the stored `indices` number of rows.
+"""
+function attach_constraint!(
+    model::JuMP.Model,
+    cons::TulipaConstraint,
+    name::Symbol,
+    container::Vector{<:JuMP.ConstraintRef},
```

Comment:
I didn't know that there is a ConstraintRef in JuMP. So, can we use [JuMP.VariableRef](https://jump.dev/JuMP.jl/stable/api/JuMP/#JuMP.VariableRef) to do the same? In that sense, we don't don't store the variables twice. Then the TulipaConstraint and TulipaVariable structure will become more reference to the model objects that make it easier to find stuff without duplicating info. What do you think?

Reply:
I didn't understand what you mean. The `TulipaVariable` should not be duplicating info, just pointing to the right place inside the model.
We could store only the name of the variable and use `model[variable.var_name]`, if that's what you mean. However, that would require passing the `model` as well everywhere, and since we are already accessing the `variables` dictionary via the variable name, it would not add anything. I.e., the key to the `TulipaVariable` inside the Dict `variables` would be stored inside the `TulipaVariable` itself, and we would still require `model` to get the actual variable. We could try making this change just to make sure that it's not creating duplicate memory, but we haven't seen such a large increase when we created the `TulipaVariable` in the past. Anyway, I digress and maybe that's not even what you mean

Reply:
I was confused for a moment, but we do have the `Vector{JuMP.VariableRef}` in the TulipaVariable. So, all good; thanks for the reply; it helped me to realize that we already have what I was trying to propose 😆

Reply:
Ah nice